File: 1222-remove-covered-intervals/remove-covered-intervals.java
class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[0] == b[0] ? b[1] - a[1] : a[0] - b[0]);

        int result = 0, lastEnd = -1;
        for (int[] interval : intervals) {
            if (lastEnd < interval[1]) {
                result++;
                lastEnd = interval[1];
            }
        }
        return result;
    }
}

File: 1222-remove-covered-intervals/remove-covered-intervals.py
'''
    1           4
            3               6
        2                           8
'''

class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x : (x[0], -x[1]))
        # print(intervals)
        result = 0
        lastEnd = 0

        # check if the current is gonna get consumed or not
        for _, end in intervals:
            if lastEnd < end:
                result += 1
                lastEnd = end

        return result


